src/inmanta/resources.py

```diff
@@ -269,11 +271,17 @@ def object_to_id(cls, model_object: DynamicProxy, entity_name: str, attribute_na
                     agent_value = agent_value[0]
 
                 agent_value = getattr(agent_value, el)
+                if type(agent_value) != str and type(agent_value) != DynamicProxy:
```

Comment:
Shouldn't we put this one after the for-loop and remove the DynamicProxy condition from the if-statement?

Reply:
This is a duplicate of Wouter his comment.